File: domain/src/main/java/com/github/dkurata38/open_web_library/domain/member/MemberCredentialId.java
package com.github.dkurata38.open_web_library.domain.member;

import com.github.dkurata38.library.domain_lib.persistence.IdIssuer;
import com.github.dkurata38.library.domain_lib.persistence.Identifier;
import lombok.Value;

@Value
public class MemberCredentialId implements Identifier<MemberCredentialId> {
	private final String value;

	private static IdIssuer<MemberCredentialId> idIssuer = new IdIssuer<>(MemberCredentialId::new);
	public static MemberCredentialId issue() {
		return idIssuer.issue();
	}

	@Override
	public Class<MemberCredentialId> getType() {
		return MemberCredentialId.class;
	}
}

File: domain/src/main/kotlin/com/github/dkurata38/open_web_library/domain/member/MemberCredentialId.kt
package com.github.dkurata38.open_web_library.domain.member

import com.github.dkurata38.library.domain_lib.persistence.IdIssuer
import com.github.dkurata38.library.domain_lib.persistence.Identifier

class MemberCredentialId(private val _value: String) : Identifier<MemberCredentialId> {

    companion object {
        private val idIssuer = IdIssuer { MemberCredentialId(it) }
        fun issue(): MemberCredentialId {
            return idIssuer.issue()
        }
    }


    override fun getType(): Class<MemberCredentialId> {
        return javaClass
    }

    override fun getValue(): String {
        return _value
    }
}
